### PR TITLE
Random number with given seed

### PR DESCRIPTION
**Pull request by licode**
also fixed bug in poisson noise.

### CODE REVIEW

**Comment by licode** [comment]
@danielballan

File: ophyd/sim.py [hunk 1]
```diff
@@ -336,6 +336,8 @@ class SynGauss(SynSignal):
     noise_multiplier : float
         Only relevant for 'uniform' noise. Multiply the random amount of
         noise by 'noise_multiplier'
+    random_state : numpy random state object
```
Comment (danielballan) [inline ophyd/sim.py]: Can you add `, optional` after the type here? While you are at it, the parameters `noise` and `noise_multiplier` should be marked optional as well.

File: ophyd/sim.py [hunk 2]
```diff
@@ -344,18 +346,19 @@ class SynGauss(SynSignal):
...
         self._motor = motor
-
+        if random_state is None:
+            random_state = np.random
         def func():
             m = motor.read()[motor_field]['value']
             v = Imax * np.exp(-(m - center) ** 2 / (2 * sigma ** 2))
             if noise == 'poisson':
-                v = int(np.random.poisson(np.round(v), 1))
+                v += int(random_state.poisson(np.round(v), 1))
```
Comment (danielballan) [inline ophyd/sim.py]: Actually, the existing code a correct implementation of Poisson noise. You can just rebase and remove the commit 28c64e6 where this was done.

File: ophyd/sim.py [hunk 3]
```diff
@@ -402,22 +407,23 @@ class Syn2DGauss(SynSignal):
...
-                 **kwargs):
+                 random_state=None, **kwargs):
 
         if noise not in ('poisson', 'uniform', None):
             raise ValueError("noise must be one of 'poisson', 'uniform', None")
         self._motor = motor0
         self._motor1 = motor1
-
+        if random_state is None:
+            random_state = np.random
```
Comment (danielballan) [inline ophyd/sim.py]: :+1:

File: ophyd/sim.py (hunk 1 above)
Comment (licode) [reply ophyd/sim.py]: good catch. I will add optional after the type.

File: ophyd/sim.py (hunk 2 above)
Comment (licode) [reply ophyd/sim.py]: 👍

**Comment by danielballan** [comment]
Note to reviewers: ensure that the stray 'tag v0.6.1' in this PR is stripped out before merging!

**Comment by jrmlhermitte** [comment]
maybe add a test as well?

**Comment by licode** [comment]
sounds good. I will add a test.

File: ophyd/tests/test_sim.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    for i in range(2):
+        s = np.random.RandomState(0)
+        noisy_det = SynGauss('noisy_det', motor, 'motor', center=0, Imax=1,
+                             noise='uniform', sigma=1, noise_multiplier=0.1,
+                             random_state=s)
+        noisy_det.trigger()
+        d = noisy_det.read()['noisy_det']['value']
+        dlist.append(d)
+    assert dlist[0] == dlist[1]
+
```
Comment (jrmlhermitte) [inline ophyd/tests/test_sim.py]: I would try a test without random_state and run
```python
for i in range(2):
    np.random.seed(0)
    noisy_det = SynGauss(...)
```
so basically try it without `random_state` flag.

Comment (licode) [reply ophyd/tests/test_sim.py]: done.

Comment (danielballan) [reply ophyd/tests/test_sim.py]: Good suggestion, @jrmlhermitte. Looks good, @licode.

**Comment by danielballan** [comment]
I'm re-running Travis. Once it's happy this is good to merge.

**Comment by jrmlhermitte** [comment]
looks good!

**Pull request merged**